src/libostree/ostree-sysroot-deploy.c

```diff
@@ -1797,14 +1798,35 @@ parse_os_release (const char *contents, const char *split)
   return ret;
 }
 
+static guint
+bootloader_get_max_boot_tries (OstreeSysroot *self, GCancellable *cancellable, GError **error)
+{
+  g_autoptr (OstreeRepo) repo = NULL;
+  if (!ostree_sysroot_get_repo (self, &repo, cancellable, error))
+    return 0;
+
+  return repo->boot_counting;
+}
+
+static gboolean
+bootloader_is_boot_count_enabled (OstreeSysroot *self, GCancellable *cancellable, GError **error)
+{
+  g_autoptr (OstreeRepo) repo = NULL;
+  if (!ostree_sysroot_get_repo (self, &repo, cancellable, error))
+    return FALSE;
+
+  return (repo->boot_counting != 0 ? TRUE : FALSE);
+}
+
 /* Generate the filename we will use in /boot/loader/entries for this deployment.
  * The provided n_deployments should be the total number of target deployments (which
  * might be different from the cached value in the sysroot).
  */
 static char *
 bootloader_entry_filename (OstreeSysroot *sysroot, guint n_deployments,
-                           OstreeDeployment *deployment)
+                           OstreeDeployment *deployment, GCancellable *cancellable, GError **error)
```

Comment:
After we drop the fallibility (use of GError) from those helpers we don't need this change either.

Reply:
sure, dropped these params